gateway/src/apicast/policy/url_rewriting/policy.lua

```diff
@@ -0,0 +1,80 @@
+--- URL rewriting policy
+-- This policy allows to modify the path of a request.
+
+local ipairs = ipairs
+local sub = ngx.re.sub
+local gsub = ngx.re.gsub
+
+local policy = require('apicast.policy')
+local _M = policy.new('URL rewriting policy')
+
+local new = _M.new
+
+-- func needs to be ngx.re.sub or ngx.re.gsub.
+-- This method simply calls one of those 2. They have the same interface.
+local function substitute(func, subject, regex, replace, options)
+  local new_uri, num_changes, err = func(subject, regex, replace, options)
+
+  if not new_uri then
+    ngx.log(ngx.WARN, 'There was an error applying the regex: ', err)
+  end
+
+  return new_uri, num_changes > 0
+end
+
+-- Returns the substitute function to be applied according to the rewrite
+-- operation. The result can be ngx.re.sub or ngx.re.gsub.
+local function substitute_func(rewrite_operation)
+  if rewrite_operation == 'sub' then
```

Comment:
Maybe better to have a dispatch table?

Reply:
OK. I don't have a strong preference about this one. I applied that pattern in the headers policy https://github.com/3scale/apicast/blob/d8ddb6a8313e22637d7b0c3bca35a3c645d106ad/gateway/src/apicast/policy/headers/policy.lua#L58

Reply:
I have a feeling it could have better performance, but can't really say without a benchmark/profile/jit trace.